File: app/src/main/java/com/proxerme/app/receiver/BootReceiver.java
package com.proxerme.app.receiver;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

import com.proxerme.app.application.MainApplication;

/**
 * Receiver for a boot. It starts and schedules all background services and tasks.
 *
 * @author Ruben Gees
 */
public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (intent.getAction().equals("android.intent.action.BOOT_COMPLETED")) {
            try {
                MainApplication application = (MainApplication) context.getApplicationContext();

                application.getNotificationManager().retrieveNewsLater(context);
                application.getNotificationManager().retrieveMessagesLater(context);
            } catch (ClassCastException e) {
                Log.e(getClass().getName(), "getApplicationContext did not return the " +
                        "Application. Notifications after boot will not work on this device.");
            }
        }
    }
}

File: app/src/main/kotlin/com/proxerme/app/receiver/BootReceiver.kt
package com.proxerme.app.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.proxerme.app.helper.ServiceHelper
import com.proxerme.app.service.ChatService

/**
 * Receiver for a boot. It starts and schedules all background services and tasks.
 *
 * @author Ruben Gees
 */
class BootReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
            ServiceHelper.retrieveNewsLater(context)
            ChatService.reschedule(context)
        }
    }
}
